src/Containers.jsx

```diff
@@ -314,45 +316,58 @@ class Containers extends React.Component {
-                    return false;
-                if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
-                    return false;
-                return true;
-            });
+            emptyCaptionPod = _("No running containers in this pod");
         }
 
-        if (this.props.textFilter.length > 0) {
-            const lcf = this.props.textFilter.toLowerCase();
-            filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
-                    this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
-            );
-        }
+        if (this.props.containers !== null && this.props.pods !== null) {
+            if (this.props.containers !== null && this.props.pods !== null)
+                filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
+
+            if (this.props.ownerFilter !== "all") {
+                filtered = filtered.filter(id => {
+                    if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
+                        return false;
+                    if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
+                        return false;
+                    return true;
+                });
+            }
 
-        filtered.sort((a, b) => {
-            // User containers are in front of system ones
-            if (this.props.containers[a].isSystem !== this.props.containers[b].isSystem)
-                return this.props.containers[a].isSystem ? 1 : -1;
-            return this.props.containers[a].Names > this.props.containers[b].Names ? 1 : -1;
-        });
+            if (this.props.textFilter.length > 0) {
+                const lcf = this.props.textFilter.toLowerCase();
+                filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
+                        this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
+                );
+            }
+
+            // Remove infra containers
+            filtered = filtered.filter(id => !this.props.containers[id].IsInfra);
 
-        const rows = filtered.map(id => this.renderRow(containersStats, this.props.containers[id], this.props.containersDetails[id]));
+            filtered.sort((a, b) => {
+                // User containers are in front of system ones
+                if (this.props.containers[a].isSystem !== this.props.containers[b].isSystem)
+                    return this.props.containers[a].isSystem ? 1 : -1;
+                return this.props.containers[a].Names > this.props.containers[b].Names ? 1 : -1;
+            });
+
+            Object.keys(this.props.pods || {}).forEach(pod => { partitionedContainers[pod] = [] });
+            filtered.forEach(id => {
```

Comment:
If I see correctly, you don't order pods. I think we want to have the same logic with pods as we have with containers - first goes user ones, then system ones and otherwise ordered by name.